dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-raft/src/main/java/org/apache/dolphinscheduler/plugin/registry/raft/EphemeralNodeManager.java

```diff
@@ -0,0 +1,405 @@
+        kvStore.bPut(Constants.MASTER_TYPE, writeUtf8(JSONUtils.toJsonString(activeNodes)));
+        //Update the mapping of the master group and master node list
+        kvStore.bPut(Constants.REGISTRY_DOLPHINSCHEDULER_MASTERS, writeUtf8(JSONUtils.toJsonString(activeNodes.keySet())));
+    }
+
+    private void updateActiveWorker(String key, Map<String, Long> activeNodes) {
+        kvStore.bPut(Constants.WORKER_TYPE, writeUtf8(JSONUtils.toJsonString(activeNodes)));
+        //Update the mapping of the worker group and worker node list
+        kvStore.bPut(key.substring(0, key.lastIndexOf(Constants.SINGLE_SLASH)), writeUtf8(JSONUtils.toJsonString(activeNodes.keySet())));
+    }
+
+    private void addDeadServer(String nodeType, String nodeAddress) {
+        final String deadServerAddress = getDeadServerSuffix(nodeType, nodeAddress);
+        List<String> deadServers = getDeadServers();
+        if (!deadServers.contains(deadServerAddress)) {
+            deadServers.add(deadServerAddress);
+            kvStore.bPut(Constants.REGISTRY_DOLPHINSCHEDULER_DEAD_SERVERS, writeUtf8(JSONUtils.toJsonString(deadServers)));
+        }
+    }
+
+    private void removeDeadServer(String nodeType, String nodeAddress) {
+        final String deadServerAddress = getDeadServerSuffix(nodeType, nodeAddress);
+        List<String> deadServers = getDeadServers();
+        if (deadServers.contains(deadServerAddress)) {
+            deadServers.remove(deadServerAddress);
+            kvStore.bPut(Constants.REGISTRY_DOLPHINSCHEDULER_DEAD_SERVERS, writeUtf8(JSONUtils.toJsonString(deadServers)));
+        }
+    }
+
+    /**
+     * @return IP:Port->TimeMillis
+     */
+    private Map<String, Long> getActiveWorkerServers() {
+        final String servers = readUtf8(kvStore.bGet(Constants.WORKER_TYPE));
+        if (StringUtils.isEmpty(servers)) {
+            return new HashMap<>();
+        }
+        return JSONUtils.toMap(servers, String.class, Long.class);
+    }
+
+    private List<String> getDeadServers() {
+        final String storedDeadServers = readUtf8(kvStore.bGet(Constants.REGISTRY_DOLPHINSCHEDULER_DEAD_SERVERS));
+        if (StringUtils.isEmpty(storedDeadServers)) {
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(JSONUtils.toList(storedDeadServers, String.class));
+    }
+
+    /**
+     * @return IP:Port->TimeMillis
+     */
+    private Map<String, Long> getActiveMasterServers() {
+        final String storedMasterServers = readUtf8(kvStore.bGet(Constants.MASTER_TYPE));
+        if (StringUtils.isEmpty(storedMasterServers)) {
+            return new HashMap<>();
+        }
+        return JSONUtils.toMap(storedMasterServers, String.class, Long.class);
+    }
+
+    private String getDeadServerSuffix(String nodeType, String serverAddress) {
+        return nodeType + Constants.UNDERLINE + serverAddress;
+    }
+
+    private void addWorkerGroup(String key) {
```

Comment:
`WorkerGroup` is belong to bussiness domain, and they are not part of the registry.